OCM-2690 | feat: Support HTPasswd from-file flag in interactive mode

ckandag opened the pull request:
https://issues.redhat.com/browse/OCM-2690

ciaranRoche reviewed (commented):
Overall looks good, I agree with Ori's comment, if the flag isnt provided we should prompt for it in interactive mode, if the user is not supplying the other args. 

Also is there anything we can do to the func `buildUserList` the amount of `if elses` can be hard to follow. Is it possible to flatten this out a little bit, by abstracting to other functions and returning early when possible. 

Also i know its not directly related to this change, but i think it would be good to add some unit tests for `parseHtpasswordFile`

ckandag commented on cmd/create/idp/htpasswd.go:
```diff
@@ -93,11 +93,32 @@ func validateUserArgs(r *rosa.Runtime) {
+
+	//interactive is enabled and  other user info is not set
+	//then prompt for the optional flag 'from-file'
+	if interactive.Enabled() &&
```
@oriAdler @ciaranRoche :  So the ask is to check for htpasswdFile == ""  before prompting,  correct?  Does the below sound better
`	if interactive.Enabled() &&
		htpasswdFile == "" && len(args.htpasswdUsers) == 0 && args.htpasswdUsername == "" {
`

oriAdler replied:
Yes, if the `htpasswdFile` was provided we can use it.
Otherwise, we can prompt the user with a message to enter the value for `htpasswdFile`.

ckandag commented:
@ciaranRoche @oriAdler  :   Can you PTAL again
- interactively prompt if filepath is not set and if other args are also set
- refactored the code to simplify the if\else structure, 
- added test for htpasswd parser

oriAdler commented on cmd/create/idp/htpasswd_test.go:
```diff
@@ -0,0 +1,113 @@
+
+func CreateTmpFile(content string) (*os.File, error) {
+	// Create a temporary file
+	file, errs := ioutil.TempFile("", "temp-*.txt")
```
```suggestion
	file, err := ioutil.TempFile("", "temp-*.txt")
```

oriAdler reviewed (commented):
Left a small comment, other than that LGTM.

oriAdler commented:
/lgtm
